js/renderCards.js

```diff
@@ -0,0 +1,148 @@
+    '[data-js="cards-unordered-list"]'
+  );
+
+  cards.forEach((card) => {
+    const answerCardInnerHtmlDiv = document.createElement("div");
+    const answerCardInnerHtml = `
+  
+      <h2 data-js="question">${card.question}</h2>
+      <div class="card__wrapper">
+      <svg
+          class="card__icon--questionmark"
+          xmlns="http://www.w3.org/2000/svg"
+          width="32"
+          height="32"
+          viewBox="0 0 512 512"
+      >
+          <path
+          fill="currentColor"
+          d="M379.063 148.114c0 89.948-99.942 83.84-99.942 146.026v.336c0 7.174-5.815 12.989-12.989 12.989h-54.53c-7.174 0-12.989-5.815-12.989-12.989v-3.668c0-96.055 87.727-89.947 87.727-134.921c0-19.433-14.436-31.093-38.311-31.093c-19.423 0-39.271 8.493-57.314 25.851c-4.624 4.448-11.854 4.729-16.907.776l-35.862-28.05c-6.044-4.727-6.745-13.685-1.402-19.193c29.94-30.865 68.846-49.343 120.369-49.343c81.618.001 122.15 43.864 122.15 93.279zm-88.838 260.403c0 27.762-22.209 50.526-50.525 50.526c-27.762 0-50.526-22.764-50.526-50.526c0-27.761 22.764-50.525 50.526-50.525c28.317 0 50.525 22.764 50.525 50.525z"
+          />
+      </svg>
+      <h2 class="card__answer hidden" data-js="answer" hidden>
+          ${card.answer}
+      </h2>
+      </div>
+      
+      `;
+    // add class
+    answerCardInnerHtmlDiv.classList.add("card__container");
+    // set inner html
+    answerCardInnerHtmlDiv.innerHTML = answerCardInnerHtml;
+    // INNER HTML for the Question and Answer Card  ----- END
+
+    //   tags li element
+    const unorderedListElement = document.createElement("ul");
+    const listItemOne = document.createElement("li");
+    listItemOne.classList.add("taglist__item");
+    listItemOne.textContent = card.tags;
+    unorderedListElement.append(listItemOne);
+
+    // main button element
+    const buttonMainElement = document.createElement("button");
+    buttonMainElement.setAttribute("data-js", "showAnswerButton");
+    buttonMainElement.classList.add("card__button");
+    buttonMainElement.textContent = "Show Answer";
+
+    // bookmark button element with logic ----------------------
+    const buttonBookmarkElement = document.createElement("button");
+    buttonBookmarkElement.classList.add("card__button--bookmark");
+    buttonBookmarkElement.setAttribute("data-js", "bookmark-button");
+    buttonBookmarkElement.addEventListener("click", () => {
+      //  Here I change the boolean value of a card obeject, then I get all the cards except the one I altered and finally I add the altered card back in via push. This approach feels wrong, but it works
+      const alteredCardObject = {
+        ...card,
+        bookmarked: !card.bookmarked,
+      };
+      const filteredArray = cards.filter(
+        (cardElement) => card.id != cardElement.id
+      );
+      filteredArray.push(alteredCardObject);
+      // const alteredCardsArray = [...cards, alteredCardObject];
+
+      localStorage.setItem("cards", JSON.stringify(filteredArray));
```

Comment:
When you want to updated an entry within an array, you can work with `.map()` maps returns you a new array with the modifications. 
You can combine it with a conditional to ensure only the element gets modified that is being clicked. 
You apparently have access to the id of the current card and can use this to identify the clicked card (otherwise you would not be able to use `.filter()` properly here).

```suggestion
      const filteredArray = cards.map((cardElement) => {
        return cardElement.id === card.id
          ? { ...card, bookmarked: !cardElement.bookmarked }
          : cardElement;
      });

      localStorage.setItem("cards", JSON.stringify(filteredArray));
```

Instead of working with a ternary operator as I did, you can also work with an if-else statement